docs/topics/async.txt

```diff
@@ -4,18 +4,110 @@ Asynchronous support
 
 .. versionadded:: 3.0
 
-Django has developing support for asynchronous ("async") Python, but does not
-yet support asynchronous views or middleware; they will be coming in a future
-release.
+Django has support for writing asynchronous ("async") views, along with an
+entirely async-enabled request stack if you are running under
+:doc:`ASGI </howto/deployment/asgi/index>` rather than WSGI. Async views will
+still work under WSGI, but with performance penalties and without the ability
+to have long-running requests efficiently.
 
-There is limited support for other parts of the async ecosystem; namely, Django
-can natively talk :doc:`ASGI </howto/deployment/asgi/index>`, and some async
-safety support.
+We're still working on asynchronous support for the ORM and other parts of
+Django; you can expect to see these in future releases. For now, you can use
+the ``sync_to_async`` adapter to interact with normal Django, as well as use a
+whole range of Python asyncio libraries natively. See below for more details.
+
+.. versionchanged:: 3.1
+
+    Support for async views was added.
+
+Async views
+===========
+
+.. versionadded:: 3.1
+
+Any view can be declared async by making the callable part of it return a
+coroutine - commonly, this is done using ``async def``. For a function-based
+view, this simply means declaring the whole view using ``async def``; for a
+class-based view, this means making its ``__call__`` method an ``async def``
+(not its ``__init__`` or ``as_view``!).
+
+.. note::
+
+    Django uses ``asyncio.iscoroutinefunction`` to test if your view is
+    asynchronous or not. If you implement your own method of returning a
+    coroutine, ensure you set the ``_is_coroutine`` attribute of the view
+    to ``asyncio.coroutines._is_coroutine`` so this function returns ``True``.
```

Comment:
(Ref @felixxm's https://github.com/django/django/pull/11650/files#r380653311, which is fine (+1).)

Are we sure we want to **document** this in Django's docs? Maybe 3.9/3.10 arrive and there's an official way to do this, and we can just point to that. 

("But we support older versions" — sure: inclined to say, "if you're using the async stuff, use the latest Python, because asyncio is still evolving rapidly"...) 

Not sure. Maybe it's OK, but what do we think?

Reply:
I want to call it out because it is an entirely non-obvious behaviour that the Python docs repeatedly gloss over. If I kept forgetting about it and needed reminding, I want our advanced users to have this told to them directly.

Reply:
OK. Makes sense.